app/util/transactions/index.js

```diff
@@ -1441,15 +1441,56 @@ export const minimumTokenAllowance = (tokenDecimals) => {
 };
 
 /**
- *
+ * For a MM Swap tx: Determines if the transaction is an ERC20 approve tx OR the actual swap tx where tokens are transferred
  */
-export const isSwapTransaction = (data, origin, to, chainId) =>
+export const getIsInSwapFlowTransaction = (data, origin, to, chainId) => {
+  if (!data) {
+    return false;
+  }
+
   // if approval data includes metaswap contract
   // if destination address is metaswap contract
-  origin === process.env.MM_FOX_CODE &&
-  to &&
-  (swapsUtils.isValidContractAddress(chainId, to) ||
-    (data &&
-      data.substr(0, 10) === APPROVE_FUNCTION_SIGNATURE &&
-      decodeApproveData(data).spenderAddress?.toLowerCase() ===
-        swapsUtils.getSwapsContractAddress(chainId)));
+  return (
+    origin === process.env.MM_FOX_CODE &&
+    to &&
+    (swapsUtils.isValidContractAddress(chainId, to) ||
+      (data &&
+        data.substr(0, 10) === APPROVE_FUNCTION_SIGNATURE &&
+        decodeApproveData(data).spenderAddress?.toLowerCase() ===
+          swapsUtils.getSwapsContractAddress(chainId)))
+  );
```

Comment:
```suggestion
      (data?.substr(0, 10) === APPROVE_FUNCTION_SIGNATURE &&
        decodeApproveData(data).spenderAddress?.toLowerCase() ===
          swapsUtils.getSwapsContractAddress(chainId)))
  );
```

(similarly just below)